constitution/actions.js

```diff
@@ -335,6 +335,47 @@ function updateServiceConfig(new_config) {
   }
 }
 
+const publicRecordsMapName = "public:records";
+const privateRecordsMapName = "records";
+const publicPrefixMapName = "public:ccf.gov.public_prefixes";
+
+function setPublicPrefix(public_prefix) {
+  // set the prefix name
+  const prefix_buf = ccf.strToBuf(public_prefix);
+  ccf.kv[publicPrefixMapName].set(prefix_buf, new ArrayBuffer(0));
+  // const prefix_len = prefix_buf.byteLength;
+
+  // copy all of the keys with the prefix over from the private map to the public one (leaving them in the private one too)
```

Comment:
Changing the history of which keys were public here seems risky/dangerous. Changing this prefix after the application has been opened is a big deal and I wouldn't expect it to be done at all for a given instance?

Reply:
I've removed the code to copy as it was already commented out. I think changing the publicity of a prefix would be ok on its own so that only new updates are affected.

Reply:
As long as it only changes the behaviour for new keys and not historical ones, I think that's fine 👍